src/AuditTrailModule/javasource/test_crm/tests/TestAuditInheritance.java

```diff
@@ -56,22 +58,166 @@ public void testCreatedRecord() throws InterruptedException, CoreException {
+		expectedLog.addLogLine(new ExpectedLogLineReferenceSet(this.context, Company.MemberNames.Company_Group.toString())
+			.keep(group.getMendixObject()).keep(group2.getMendixObject()));
+
+		final ActualLog actualLog = new ActualLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+
+	@Test
+	public void testDeleteRecordWithReference() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany(group, group2);
+
+		company.delete();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = new ExpectedLog(0, TypeOfLog.Delete, Company.entityName, adminId, initialDate, Company.entityName);
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.Name.toString(), NAME, NAME));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.CompanyNr.toString(), COMPANY_NR, COMPANY_NR));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.InternNr.toString(), company.getInternNr().toString(), company.getInternNr().toString()));
+		expectedLog.addLogLine(new ExpectedLogLineReferenceSet(this.context, Company.MemberNames.Company_Group.toString())
+			.keep(group.getMendixObject()).keep(group2.getMendixObject()));
+
 		final ActualLog actualLog = new ActualLog(context, company.getMendixObject().getId().toLong());
 		expectedLog.verify(actualLog);
 	}
 	
-	private Company createBaseCompany() throws CoreException {
+
+	@Test
+	public void testRemoveReference() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany(group, group2);
+
+		company.setCompany_Group(Collections.emptyList());
+		company.commit();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = new ExpectedLog(1, TypeOfLog.Change, Company.entityName, adminId, initialDate, Company.entityName);
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.Name.toString(), NAME, NAME));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.CompanyNr.toString(), COMPANY_NR, COMPANY_NR));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.InternNr.toString(), company.getInternNr().toString(), company.getInternNr().toString()));
+		expectedLog.addLogLine(new ExpectedLogLineReferenceSet(this.context, Company.MemberNames.Company_Group.toString())
+			.delete(group.getMendixObject()).delete(group2.getMendixObject()));
+
+		final ActualLog actualLog = new ActualLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+
+	@Test
+	public void testChangeReference() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany(group);
+
+		company.setCompany_Group(Arrays.asList(group2));
```

Comment:
Since we have only one element in the collection I think it's better to use `Collections.singletonList(group2)`.

Reply:
Absolutely. Done.